1.10/joint.sdf: better default limit values

scpeters opened the pull request:
# 🦟 Bug fix

Part of #169.

## Summary

* Use `±inf` instead of `±1e16` for position limits
* Use `inf` instead of `-1` for effort, velocity limits

This will require a small change in gz-physics, as illustrated by the test failure in https://github.com/gazebosim/gz-physics/pull/401.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

chapulina commented on src/JointAxis.cc:
```diff
@@ -142,11 +143,12 @@ Errors JointAxis::Load(ElementPtr _sdf)
-    this->dataPtr->upper = limitElement->Get<double>("upper", 1e16).first;
-    this->dataPtr->effort = limitElement->Get<double>("effort", -1).first;
+    const double kInf = std::numeric_limits<double>::infinity();
+    this->dataPtr->lower = limitElement->Get<double>("lower", -kInf).first;
```
You could pass the variable itself here as the default value, so if it ever changes again, it only needs to be changed on line 64, i.e.

```suggestion
    this->dataPtr->lower = limitElement->Get<double>("lower", this->dataPtr->lower).first;
```

scpeters replied:
I've deduplicated the default values used in `JointAxis::Load` in https://github.com/gazebosim/sdformat/pull/1112/commits/c7feca99e73fdc684a9bc4705510274294d09ef5

scpeters commented:
while deduplicating some default value constants, I noticed an unused variable corresponding to the deprecated and removed `use_parent_model_frame` element, so I removed that as well in https://github.com/gazebosim/sdformat/pull/1112/commits/95b3963d430eda20c301135dc53f3f2022d34710

azeey commented on sdf/1.10/joint.sdf:
```diff
@@ -74,16 +74,16 @@
         <description>Specifies the upper joint limit (radians for revolute joints, meters for prismatic joints). Omit if joint is continuous.</description>
       </element>
-      <element name="effort" type="double" default="-1" required="0">
+      <element name="effort" type="double" default="inf" required="0">
```
The description says "Limit is not enforced if value is negative". Something to 
keep in mind when updating gz-physics.

scpeters replied:
this was already part of the behavior of the DOM object as of https://github.com/gazebosim/sdformat/pull/357

azeey commented:
We'll also need to update sdformat-mjcf https://github.com/gazebosim/gz-mujoco/blob/fd5dec325052b0db93fadc88e667987d27bf378b/sdformat_mjcf/src/sdformat_mjcf/sdformat_to_mjcf/converters/joint.py#L29

scpeters commented:
> We'll also need to update sdformat-mjcf https://github.com/gazebosim/gz-mujoco/blob/fd5dec325052b0db93fadc88e667987d27bf378b/sdformat_mjcf/src/sdformat_mjcf/sdformat_to_mjcf/converters/joint.py#L29

opened a PR https://github.com/gazebosim/gz-mujoco/pull/106

but its tests are failing
